dotnet/src/SemanticKernel.Abstractions/Orchestration/KernelResult.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Kernel result after execution.
+/// </summary>
+public sealed class KernelResult
```

Comment:
Should `FunctionResult` and `KernelResult` shared a common base class?

Reply:
Initially, I tried this approach, but then I decided to remove base class, since it introduces some coupling between FunctionResult and KernelResult, which I would like to avoid. I think these types shouldn't be related to each other, even if they have some common properties or methods. 

For common methods like `GetValue` we can extract it to some static class and re-use in both classes if needed, without introducing base class. 
In terms of fields like `Value` or `ModelResults`, again, not sure that moving them to the base class will introduce significant benefits, but on the other hand, coupling may introduce some issues later.